kalite/contentload/management/commands/generate_assessment_zips.py

```diff
@@ -100,25 +106,100 @@ def all_image_urls(items):
 
     return newitems
 
-def convert_urls(string):
-    """ Convert urls in i18n strings into localhost urls. """
-    url_to_replace = r'https?://[\w\.\-\/]+\/(?P<filename>[\w\.\-]+\.(png|gif|jpg|JPEG|JPG))'
-    return re.sub(url_to_replace, _old_item_url_to_content_url, string)
 
-def _old_item_url_to_content_url(matchobj):
+def localize_image_urls(item_data):
+
+    url_to_replace = r'https?://[\w\.\-\/]+\/(?P<filename>[\w\.\-]+\.(png|gif|jpg|jpeg))'
+
+    return re.sub(url_to_replace, _old_image_url_to_content_url, item_data, flags=re.IGNORECASE)
+
+
+def convert_urls(item_data):
+    """Convert urls in i18n strings into localhost urls.
+    This function is used by ka-lite-central/centralserver/i18n/management/commands/update_language_packs.py"""
+    item_data = localize_image_urls(item_data)
+    item_data = localize_content_links(item_data)
+    return item_data
+
+
+def _old_image_url_to_content_url(matchobj):
     return "/content/khan/%s" % matchobj.group("filename")
+
+
+def localize_all_content_links(items):
+    # we copy so we make sure we don't modify the items passed in to this function
+    newitems = copy.deepcopy(items)
+
+    # loop over every item, and replace links in them to point to local resources, if available
+    for item in newitems.itervalues():
+        item['item_data'] = localize_content_links(item['item_data'])
+
+    return newitems
+
+
+def localize_content_links(item_data):
+    # this ugly regex looks for links to content on the KA site, also including the markdown link text and surrounding bold markers (*), e.g.
+    # **[Read this essay to review](https://www.khanacademy.org/humanities/art-history/art-history-400-1300-medieval---byzantine-eras/anglo-saxon-england/a/the-lindisfarne-gospels)**
+    # TODO(jamalex): answer any questions people might have when this breaks!
+    url_pattern = r"(?P<prefix>\**\[[^\]\[]+\]\()https?://www\.khanacademy\.org/[\/\w\-]*/./(?P<slug>[\w\-]+)(?P<suffix>\)\**)"
+
+    return re.sub(url_pattern, _old_content_links_to_local_links, item_data)
+
+
+def _old_content_links_to_local_links(matchobj):
+    # replace links in them to point to local resources, if available, otherwise return an empty string
+    content = _get_content_by_readable_id(matchobj.group("slug"))
+    if not content or "path" not in content:
+        print "Content link target not found:", matchobj.group()
+        return ""
+
+    return "%s/learn/%s%s" % (matchobj.group("prefix"), content["path"], matchobj.group("suffix"))
+
+
+CONTENT_BY_READABLE_ID = None
+def _get_content_by_readable_id(readable_id):
+    global CONTENT_BY_READABLE_ID
+    if not CONTENT_BY_READABLE_ID:
+        CONTENT_BY_READABLE_ID = dict([(c["readable_id"], c) for c in get_content_cache().values()])
+    try:
+        return CONTENT_BY_READABLE_ID[readable_id]
+    except KeyError:
+        return CONTENT_BY_READABLE_ID.get(re.sub("\-+", "-", readable_id).lower(), None)
+
+
+def list_all_exercises_with_bad_links():
```

Comment:
This function is quite heavy. This also seems to be a "one-off" function not meant to be used anywhere. Can you put a leading underscore to indicate that it's an internal function? Also gives us an excuse not to write a test for it.